EIPS/eip-3068.md

```diff
@@ -0,0 +1,297 @@
+---
+eip: 3068
+title: Precompile for BN256 HashToCurve Algorithms
+author: Dr. Christopher Gorman (@chgormanMH)
+discussions-to: https://ethereum-magicians.org/t/pre-compile-for-bls/3973
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-10-23
+requires: EIP-198, EIP-1108
+---
+
+## Simple Summary
+This EIP defines a hash-to-curve precompile for use in BN256
+and would allow for cheaper BLS signature verification.
+
+## Abstract
+There is currently no inexpensive way to perform BLS signature
+verification for arbitrary messages.
+This stems from the fact that there is no precompiled contract
+in the EVM for a hash-to-curve algorithm for the BN256 elliptic curve.
+The gas cost of calling a deterministic hash-to-curve algorithm
+written in Solidity is approximately that of one pairing check,
+although the latter requires an order of magnitude
+more computation.
+This EIP remedies this by implementing a hash-to-curve algorithm
+for the BN256 G1 curve, which would reduce the cost of
+signature verification to essentially that of the pairing check
+precompiled contract.
+We also include a hash-to-curve algorithm for the BN256 G2 group.
+
+## Motivation
+The precompiled contracts in
+[EIP-198](./eip-198.md) and
+[EIP-1108](./eip-1108.md)
+increased usage of cryptographic operations in the EVM
+by reducing the gas costs.
+In particular, the cost reduction from
+[EIP-1108](./eip-1108.md)
+helps increase the use of SNARKs in Ethereum
+via an elliptic curve pairing check;
+however, a hash-to-curve algorithm enabling arbitrary
+BLS signature verification on BN256 in the EVM was noticeably missing.
+There is interest in having a precompiled contract which would allow
+for signature verification, as noted
+[here](https://ethereum-magicians.org/t/pre-compile-for-bls/3973).
+
+At this time, we are able to perform addition, scalar multiplication,
+and pairing checks in BN256.
+Reducing these costs in
+[EIP-1108](./eip-1108.md)
+made [ETHDKG](https://github.com/PhilippSchindler/ethdkg),
+a distributed key generation protocol in Ethereum,
+less expensive.
+ETHDKG by itself is useful; however, what it is lacking is
+the ability to verify arbitrary BLS signatures.
+Creating group signatures by aggregating partial signatures
+is one goal of a DKG protocol.
+The DKG enables the computation of partial signatures to be
+combined into a group signature offline, but there is no
+easy way to verify partial signatures or group signatures
+in the EVM.
+
+In order to perform BLS signature validation, a hash-to-curve
+algorithm is required.
+While the MapToGroup method initially discussed in the original BLS
+[paper](../assets/eip-3068/weilsigs.ps)
```

Comment:
This doesn't appear to be a human readable document.  Also, it completely chokes the spellchecker that is blocking merging this EIP as draft.  Can a human readable version be included instead?

Reply:
I converted the PostScript file to PDF. Hopefully that fixes things.